.github/workflows/compile-examples.yml

```diff
@@ -47,6 +49,27 @@ jobs:
           - fqbn: arduino:mbed_opta:opta
             platforms: |
               - name: arduino:mbed_opta
+        include:
+          - board:
+            - fqbn: arduino:mbed_portenta:envie_m7
+              additional-sketch-paths: |
+                - examples/SimpleStorageWriteRead
+                - examples/PortentaH7Logger
+                - examples/AdvancedUSBInternalOperations
+                - examples/BackupInternalPartitions
+          - board:
+            - fqbn: arduino:renesas_portenta:portenta_c33
+              additional-sketch-paths: |
+                - examples/SimpleStorageWriteRead
+                - examples/AdvancedUSBInternalOperations
+                - examples/BackupInternalPartitions
+          - board:
+            - fqbn: arduino:mbed_opta:opta
+              additional-sketch-paths: |
+                - examples/SimpleStorageWriteRead
+                - examples/AdvancedUSBInternalOperations
+                - examples/BackupInternalPartitions
```

Comment:
```suggestion
              fqbn: arduino:mbed_portenta:envie_m7
            additional-sketch-paths: |
              - examples/SimpleStorageWriteRead
              - examples/PortentaH7Logger
              - examples/AdvancedUSBInternalOperations
              - examples/BackupInternalPartitions
          - board:
              fqbn: arduino:renesas_portenta:portenta_c33
            additional-sketch-paths: |
              - examples/SimpleStorageWriteRead
              - examples/AdvancedUSBInternalOperations
              - examples/BackupInternalPartitions
          - board:
              fqbn: arduino:mbed_opta:opta
            additional-sketch-paths: |
              - examples/SimpleStorageWriteRead
              - examples/AdvancedUSBInternalOperations
              - examples/BackupInternalPartitions
```

Fix the data structure of [`jobs.<job_id>.strategy.matrix.include`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrixinclude)

---

Alternatively, you can just add the `additional-sketch-paths` data directly under the matrix elements in `jobs.build.strategy.matrix.board`. `jobs.<job_id>.strategy.matrix.include` is useful in cases where you want to apply the same sub-configuration to the matrix jobs of multiple boards, but in this case where you are configuring each matrix job individually, there is no benefit to balance out the increased complexity of the workflow code.